qa/workunits/fs/fscrypt_cli_setup.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+echo "Custom fscrypt CLI setup begin"
+WORK_DIR=$(pwd)
+set -xe
+
+# Check and print kernel encryption support
+zgrep -h ENCRYPTION /proc/config.gz /boot/config-$(uname -r) | sort | uniq || true
+cat /proc/mounts || true
+
+# Update the package manager cache
+if command -v apt-get &>/dev/null; then
+    sudo apt-get update -y
+elif command -v yum &>/dev/null; then
+    sudo yum makecache --refresh -y
+else
+    echo "Unsupported package manager. Exiting."
+    exit 1
+fi
+
+# Remove any pre-installed fscrypt packages
+if command -v apt-get &>/dev/null; then
+    sudo apt-get remove --purge -y fscrypt || true
+elif command -v yum &>/dev/null; then
+    sudo yum remove -y fscrypt || true
+fi
+
+# Check if /tmp/fscrypt exists
+if [ -d "$WORK_DIR/fscrypt" ]; then
+    echo "$WORK_DIR/fscrypt exists. Cleaning up..."
+
+    # Navigate to the folder
+    cd "$WORK_DIR/fscrypt" || { echo "Failed to navigate to $WORK_DIR/fscrypt"; exit 1; }
+
+    # Uninstall the existing installation
+    if [ -f "Makefile" ]; then
+        echo "Running 'sudo make uninstall'..."
+        sudo make uninstall || echo "Uninstall failed or no uninstall target defined."
+    else
+        echo "Makefile not found. Skipping 'make uninstall'."
+    fi
+
+    # Navigate back
+    cd || { echo "Failed to navigate back to home directory"; exit 1; }
+
+    # Delete the directory
+    echo "Deleting $WORK_DIR/fscrypt..."
+    rm -rf "$WORK_DIR/fscrypt"
+    echo "$WORK_DIR/fscrypt cleanup complete."
+else
+    echo "$WORK_DIR/fscrypt does not exist. No action needed."
+fi
+
+# Install required dependencies for building fscrypt
+if command -v apt-get &>/dev/null; then
+    sudo apt-get install -y build-essential libpam0g-dev libtinfo-dev golang
+elif command -v yum &>/dev/null; then
+    sudo yum install -y gcc make pam-devel ncurses-devel golang
+fi
+
+# Clone the custom fscrypt repository, build, and install
+git clone https://github.com/salieri11/fscrypt.git -b wip-ceph-fuse "$WORK_DIR/fscrypt"
```

Comment:
We shouldn't tie this to a personal repo, what about storing it on git.ceph.com instead?

Reply:
I will try it, thanks, didn't know it exists...

Reply:
I asked on Slack channel, there is no format process for adding new project to that git, and they suggested to send the description and justification so someone will evaluate it. Sent the info

Reply:
Nothing new by now. I suggest to keep it as is and if they will respond, we can fix it

Reply:
The goal of this is to prepare for testing userspace fscrypt. I don't want to put something temp like this in upstream. If we don't get an answer soon we'll put this in wip-fscrypt and fix it before that PR is merged.